scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/Synchronization.java

```diff
@@ -0,0 +1,1135 @@
+     * remapping function, or removes if the result is {@code null}. This
+     * method may be of use when combining multiple mapped values for a key.
+     *
+     * The function is passed as a String containing a groovy closure definition which matches the {@link java.util.function.BiFunction} specification.
+     * The first parameter receives the key, and the second parameter receives the binding associated with this key.
+     *
+     * For example, to either create or append a {@code String msg} to a
+     * value mapping:
+     *
+     * <pre> {@code
+     * store.merge(channel, key, msg, "{k, x -> x.concat(\"" + msg + "\")}")
+     * }</pre>
+     *
+     * <p>If the function returns {@code null} the mapping is removed.  If the
+     * function itself throws an (unchecked) exception, the exception is
+     * rethrown, and the current mapping is left unchanged.
+     *
+     * The default implementation is equivalent to performing the following
+     * steps for this {@code map}, then returning the current value or
+     * {@code null} if absent:
+     *
+     * <pre> {@code
+     * V oldValue = map.get(key);
+     * V newValue = (oldValue == null) ? value :
+     *              remappingFunction.apply(oldValue, value);
+     * if (newValue == null)
+     *     map.remove(key);
+     * else
+     *     map.put(key, newValue);
+     * }</pre>
+     *
+     * <p>This implementation guarantees that the operation is performed atomically on the channel map.
+     *
+     * @param channel channel identifier
+     * @param key key with which the resulting value is to be associated
+     * @param value the non-null value to be merged with the existing value
+     *        associated with the key or, if no existing value or a null value
+     *        is associated with the key, to be associated with the key
+     * @param remappingFunction the function to recompute a value if present
+     * @return the new value associated with the specified key, or null if no
+     *         value is associated with the key
+     * @throws NullPointerException if the specified key is null and this map
+     *         does not support null keys or the value or remappingFunction is
+     *         null
+     * @throws CompilationException if an error occurred during the compilation of the specified action
+     * @throws ClosureEvaluationException if an error occurred during the evaluation of the specified action
+     * @throws InvalidChannelException if the channel does not exist
+     * @throws IOException if an error occurred when persisting the modified channel on disk
+     * @since 1.8
+     * @see java.util.Map#merge(Object, Object, BiFunction)
+     * @see java.util.function.BiFunction
+     * @see <a href="http://groovy-lang.org/closures.html">Groovy Closures</a>
+     */
+    Serializable merge(String channel, String key, Serializable value, String remappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)